scanpipe/pipes/d2d.py

```diff
@@ -774,94 +777,163 @@ def _map_javascript_resource(
-        return
 
-    filename = package_data.get("filename")
-    if not filename:
-        # Cannot map anything without the about_resource value.
-        project.add_error(
-            description="ABOUT file does not have about_resource",
-            model="map_about_files",
-            details=error_message_details,
-        )
-        return
+    def match_to_resources(self, to_resources):
+        for to_resource in to_resources:
+            resource_matched = False
+            for about_path, regex_pattern in self.regex_by_about_path.items():
+                if regex_match(pattern=regex_pattern, string=to_resource.path):
+                    resource_matched = True
+                    break
 
-    ignored_resources = []
-    if extra_data := package_data.get("extra_data"):
-        ignored_resources = extra_data.get("ignored_resources")
-
-    # Fetch all resources that are covered by the .ABOUT file.
-    codebase_resources = to_resources.filter(path__contains=f"/{filename.lstrip('/')}")
-    if not codebase_resources:
-        # If there's nothing to map on the ``to/`` do not create the package.
-        project.add_warning(
-            description=(
-                "Resource paths listed at about_resource is not found"
-                " in the to/ codebase"
-            ),
-            model="map_about_files",
-            details=error_message_details,
-        )
-        return
+            if not resource_matched:
+                continue
 
-    # Ignore resources for paths in `ignored_resources` attribute
-    if ignored_resources:
-        lookups = Q()
-        for resource_path in ignored_resources:
-            lookups |= Q(**{"path__contains": resource_path})
-        codebase_resources = codebase_resources.filter(~lookups)
+            ignore_regex_patterns = self.ignore_regex_by_about_path.get(about_path, [])
+            ignore_resource = False
+            for ignore_regex_pattern in ignore_regex_patterns:
+                if regex_match(pattern=ignore_regex_pattern, string=to_resource.path):
+                    ignore_resource = True
+                    break
 
-    # Create the Package using .ABOUT data and assigned related codebase_resources
-    pipes.update_or_create_package(project, package_data, codebase_resources)
+            if ignore_resource:
+                continue
 
-    # Map the .ABOUT file resource to all related resources in the ``to/`` side.
-    for to_resource in codebase_resources:
-        pipes.make_relation(
-            from_resource=about_file_resource,
-            to_resource=to_resource,
-            map_type="about_file",
-        )
+            mapped_resources_about = self.mapped_resources_by_aboutpath.get(about_path)
+            if mapped_resources_about:
+                mapped_resources_about.append(to_resource)
+            else:
+                self.mapped_resources_by_aboutpath[about_path] = [to_resource]
+            to_resource.update(status=flag.ABOUT_MAPPED)
```

Comment:
This is used to update resources elsewhere in d2d and other pipelines too. See https://github.com/nexB/scancode.io/blob/main/scanpipe/pipes/d2d.py#L284 for example

Reply:
Actually we have the UpdateMixin at https://github.com/nexB/scancode.io/blob/main/scanpipe/models.py#L429 which defines an update() method to use everywhere, so instead of doing a
```
to_resource.status = flag.ABOUT_MAPPED
to_resource.save(update_fields=["status"])
```
we now do
```
to_resource.update(status=flag.ABOUT_MAPPED)
```
which takes less space, is cleaner and better. But both essentially do the same thing as per the update() method created. The CodebaseResource class inherits from the ProjectRelatedModel class, which inherits the Updatemixin class with this method, so this is indeed what is being used here.
So I think we are good here.